shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/messaging/MessageReaderAdapter.java

```diff
@@ -74,6 +76,10 @@ private MetaStoreEvent map(ListenerEvent listenerEvent) {
       break;
     case ALTER_PARTITION:
       AlterPartitionEvent alterPartition = (AlterPartitionEvent) listenerEvent;
+
+      if (alterPartition.getPartitionLocation().equals(alterPartition.getOldPartitionLocation())) {
```

Comment:
this `getPartitionLocation` can be null right?

Reply:
Theoretically its not possible because we have a null check on the partition here: https://github.com/ExpediaInc/apiary-extensions/blob/master/apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java#L169

I can't think of a situation where a partition is not on a location. Have you seen such a situation ?

Reply:
current situation, messages are sent without a location so they will be null

Reply:
If that's the case then we should use `Objects.equals()` here so we don't have to faff around with null checks.

Reply:
I just don't want to tie us to a listener version also deployment will be hard because current messages in the queue don't have a location. Will be easier if we gracefully handle it.

Reply:
`Objects.equals` won't be enough for our case because we don't want to do a `METADATA_UPDATE` when `alterPartition.getPartitionLocation() == null`.

So, added an explicit null check before the `.equals()` check.